File: src/main/java/net/atos/client/or/object/ObjectsClientService.java
/*
 * SPDX-FileCopyrightText: 2021 Atos, 2025 INFO.nl
 * SPDX-License-Identifier: EUPL-1.2+
 */
package net.atos.client.or.object;

import java.util.UUID;

import jakarta.enterprise.context.ApplicationScoped;
import jakarta.inject.Inject;

import org.eclipse.microprofile.rest.client.inject.RestClient;

import nl.info.client.or.objects.model.generated.ModelObject;

@ApplicationScoped
public class ObjectsClientService {
    private ObjectsClient objectsClient;

    /**
     * Default no-arg constructor, required by Weld.
     */
    public ObjectsClientService() {
    }

    @Inject
    public ObjectsClientService(
            @RestClient ObjectsClient objectsClient
    ) {
        this.objectsClient = objectsClient;
    }

    /**
     * Read {@link ModelObject} via its UUID.
     * Throws a RuntimeException if the {@link ModelObject} can not be read.
     *
     * @param object UUID of {@link ModelObject}.
     * @return {@link ModelObject}. Never 'null'!
     */
    public ModelObject readObject(final UUID object) {
        return objectsClient.objectRead(object);
    }
}

File: src/main/kotlin/nl/info/client/or/object/ObjectsClientService.kt
/*
 * SPDX-FileCopyrightText: 2021 Atos, 2025 INFO.nl
 * SPDX-License-Identifier: EUPL-1.2+
 */
@file:Suppress("PackageName")

package nl.info.client.or.`object`

import jakarta.enterprise.context.ApplicationScoped
import jakarta.inject.Inject
import nl.info.client.or.objects.model.generated.ModelObject
import nl.info.zac.util.AllOpen
import nl.info.zac.util.NoArgConstructor
import org.eclipse.microprofile.rest.client.inject.RestClient
import java.util.UUID

@ApplicationScoped
@NoArgConstructor
@AllOpen
class ObjectsClientService @Inject constructor(
    @RestClient private val objectsClient: ObjectsClient
) {
    fun readObject(objectUUID: UUID): ModelObject = objectsClient.objectRead(objectUUID = objectUUID)
}
